### PR TITLE
Support for Cognito.SignUp recognition

### PR DESCRIPTION
**Pull request by zgibek**

### CODE REVIEW

**Comment by audalykh** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: src/main/java/com/smockin/mockserver/engine/MockedRestServerEngineUtils.java [hunk 1]
```diff
@@ -344,7 +354,16 @@ private String determineAwsAccessKeyBasedOnRequest(Map<String, String> headers)
     }
 
     private boolean isAwsServiceCall(HttpClientCallDTO httpClientCallDTO) {
-        final String authHeader = httpClientCallDTO.getHeaders().get(HttpHeaders.AUTHORIZATION);
+        final Map<String, String> headers = httpClientCallDTO.getHeaders();
+        final String xAmzTargetHeader = AWS4Signer.getHeader(headers, AWS4SignerBase.HEADER_X_AMZ_TARGET);
+        if (X_AMZ_TARGET_COGNITO_SIGNUP.equals(xAmzTargetHeader)
+            || X_AMZ_TARGET_COGNITO_INITIATE_AUTH.equals(xAmzTargetHeader))
+        {
```
Comment (audalykh) [inline src/main/java/com/smockin/mockserver/engine/MockedRestServerEngineUtils.java:361]: :warning: **Braces Should Follow The "Egyptian" Style**  
  
Java standards for formatting supports the usage of braces in [Egyptian style](https://raw.githubusercontent.com/LeonardAnghel/CleanCode/master/Egyptian%20braces.png). In other words, Java standards statement is "[No line break before the opening brace](https://google.github.io/styleguide/javaguide.html#s4.1-braces)". Consider adopting this standard at the project level.

**[More details](https://confluence.devfactory.com/x/GQfhEw)**

## The improvement suggestion in this case

```java
    || X_AMZ_TARGET_COGNITO_INITIATE_AUTH.equals(xAmzTargetHeader)) {
```

[//]: # (<a id="0cc86d18-3d83-4481-a708-d5f989aa9e5e" data-finding-id="333514521" href="none" codereview-uuid-reserved-tag data-business-id="b91c1477-07d3-4cdf-b611-ba09e792e8d7" data-analyzer="java" />)

**Review by audalykh — COMMENTED** [review]
Hi, I am the Java expert who performed the review of this Pull Request. We have well-defined [coding and review standards](https://confluence.devfactory.com/display/CodeReview/Findings), and you will find below suggestions to improve the code quality, based on these guidelines.
My additional comments are in the Notes section. I hope you find them useful.

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  0</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:<b>Yellows:</b>  1</td>
    <td>:white_check_mark:<b>Greens:</b>  0</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td><b>Severe Findings</b> point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td><b>Red Findings</b> represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td><b>Yellows</b> are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR, and should not be used for FTAR calculation.
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td><b>Green Findings</b> highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td><b>Repeated</b> instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 75.00 % to 80.00 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>80.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>5</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>4</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by audalykh** [comment]
Hi, I'm Alexey Udalykh from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/devfactory-aws-mocking-sandbox/pull/7&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Comment by audalykh** [comment]
:white_check_mark: No new issues have been introduced since the previous review.

**Good work!**
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 75.00 % to 80.00 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>80.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>5</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>4</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Comment by emil-cazacu-aurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

Please consider using [Analyzer UI](https://confluence.devfactory.com/display/CodeReview/Analyzer+UI) extension to catch issues in pull requests.

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

**Comment by emil-cazacu-aurea** [comment]
:white_check_mark: No new issues have been introduced since the previous review.

**Good work!**
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 75.00 % to 80.00 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>80.00 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>5</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>4</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>1</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Review by fogone — APPROVED** [review]
LGTM

**Pull request merged**